### DISCUSSION TITLE
Adding Picture-in-picture mode to Devtools

### DISCUSSION THREAD

**Discussion by zhondori** [TanStack/query#6009] [Ideas]
### [https://developer.chrome.com/docs/web-platform/document-picture-in-picture/](https://developer.chrome.com/docs/web-platform/document-picture-in-picture/)

**Comment by TkDodo** [comment]
This looks great. @ardeora this would be a great feature for v5 🚀

**Reply by ardeora** [reply]
Took a look at this today. Unfortunately it's a bit more complex than the docs linked above


https://github.com/TanStack/query/assets/45807386/4f3bb0e1-e649-42ff-bb0f-672d0b7a4e1e

This mostly works out of the box but the Picture in Picture Window loses all the event listeners from the interactive elements! Currently the only way to add interactivity back is to add each listener back manually in Picture-In-Picture mode. I also tried to replicate this with a less complex UI (React) with a single button, and saw the same issue. Also it's a relatively new API so not a lot of docs and resources to find workarounds to this.

**Reply by TkDodo** [reply]
Would still be worth it to have them popped out readonly imo
